e2e/test/MessageSendTests.cs

```diff
@@ -496,6 +499,49 @@ public async Task X509_DeviceSendSingleMessage_Http()
         {
             await SendSingleMessage(TestDeviceType.X509, Client.TransportType.Http1).ConfigureAwait(false);
         }
+        
+        [TestMethod]
+        public async Task MultiplexingTest_Amqps()
+        {
+            await SendMessagesOverMultiplex(Client.TransportType.Amqp_Tcp_Only, MultiplexingDeviceCount).ConfigureAwait(true);
+        }
+
+        [TestMethod]
+        public async Task MultiplexingTest_Amqps_Ws()
+        {
+            await SendMessagesOverMultiplex(Client.TransportType.Amqp_WebSocket_Only, MultiplexingDeviceCount).ConfigureAwait(true);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceClientCanBeClosedAndReOpened()
+        {
+            foreach (Client.TransportType transportType in Enum.GetValues(typeof(Client.TransportType)))
+            {
+                await DeviceClientCanBeClosedAndReOpened(3, transportType).ConfigureAwait(true);
+            }
+        }
+
+        private async Task DeviceClientCanBeClosedAndReOpened(int OpenAndCloseCount, Client.TransportType transportType)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+
+            for (int attempt = 0; attempt < OpenAndCloseCount; attempt++)
```

Comment:
I see that you are testing this in serial, what happens in case where it is not serial ? isn't it one of the issue as well ? Please add a test case for that.

Reply:
I have added a test case for doing this in parallel now as well